docs/modules/ROOT/pages/documentation/ecosystem/docker-images/server-image-overview.adoc

```diff
@@ -0,0 +1,214 @@
+|`SCRIPT_DIR`| The directory where the `generate_deploy_commands.sh` and `startInForeground.sh` scripts can be found| `/opt/payara/scripts` 
+|`CONFIG_DIR`| The directory where the post and pre boot files are generated to by default| `/opt/payara/config` 
+|`DEPLOY_DIR`| The directory where applications are searched for in `generate_deploy_commands.sh` script| `/opt/payara/deployments`
+|`ADMIN_USER`| The default username credential for the default administrator user| `admin`
+|`ADMIN_PASSWORD`| The default password credential for the default administrator user. Can only be set when the basic image is created| `admin`
+|`PASSWORD_FILE`| The location of the password file for asadmin. This can be passed to asadmin using the `--passwordfile` parameter| `/opt/payara/passwordFile`
+|`DOMAIN_NAME`| The name of the domain running within the container | `domain1`
+|`AS_ADMIN_MASTERPASSWORD`| The master password to pass to Payara Server. This is overridden if one is specified in the `$PASSWORD_FILE`| `changeit`
+|===
+
+[[executing-admin-commands-domain-startup]]
+=== Executing Administration Commands at Domain Startup
+
+It’s possible to run a set of custom administration commands during the domain startup. You can either specify the `PREBOOT_COMMANDS` or `POSTBOOT_COMMANDS` environment variables to point to the absolute path of a custom script file, or you can just copy the file to the expected paths (See above).
+
+For example, the following sample command will execute all commands listed in the `post-boot-commands.asadmin` file inside the `/local/path/with/boot/file` directory mounted volume:
+
+[source, shell]
+----
+docker run -p 8080:8080 -v /local/path/with/boot/file:/config -e POSTBOOT_COMMANDS=/config/post-boot-commands.asadmin payara/server-full
+----
+
+Alternatively, the same outcome can be achieved by defining a custom Docker image:
+
+[source, Docker]
+----
+FROM payara/server-full
+COPY post-boot-commands.asadmin $POSTBOOT_COMMANDS
+----
+
+[[executing-custom-scripts-before-domain-startup]]
+=== Executing Custom Scripts before Domain Startup
+
+If preboot command files are not enough, you can add your own customized shell scripts to the `${SCRIPT_DIR}` directory, which will allow you to modify the environment before the Payara Server domain starts. 
+
+Scripts added to this folder should be named with the following convention:
+
+----
+init_<num>_<text>.sh 
+----
+* `<num>` is a positive integer that allows you to set the order in which the scripts are run.
+* `<text>` is a descriptive help text that summarizes what the script does.
+
+NOTE: Be aware that the default deploy command script (`init_1_generate_deploy_commands.sh`) also uses follows this convention, so custom scripts should start with number 2.
+
+As with the preboot command file, you may either create a custom Docker image that already contains the script files in question or you can also mount a volume mapped to the `${SCRIPT_DIR}` directory instead.
```

Comment:
I would only document putting the custom scripts into the config/init.d directory and not directly to the config directory. It's cleaner and also sufficient. I propose to change this whole section as follows:

```suggestion
If preboot command files are not enough, you can add your own customized shell scripts to the `${SCRIPT_DIR}/init.d` directory. This will allow you to modify the environment before the Payara Server domain starts. 

Scripts added to this folder should be https://www.gnu.org/software/bash/[Bash] scripts and they should end with the `.sh` suffix.

TIP: Scripts in `${SCRIPT_DIR}/init.d` will be executed in the standard file order. If you need to start them in specific order, we recommend to prepend a number prefix to their name, for example `01_script1.sh`, `02_script2.sh`.

As with the preboot command file, you may either create a custom Docker image that already contains the script files in question or you can also mount a volume mapped to the `${SCRIPT_DIR}/init.d` directory instead.
```

Reply:
I agree with your suggestion, @OndroMih, but the current script in the image allows both options:

```
for f in ${SCRIPT_DIR}/init_* ${SCRIPT_DIR}/init.d/*; do
      case "$f" in
        *.sh)  echo "[Entrypoint] running $f"; . "$f" ;;
        *)     echo "[Entrypoint] ignoring $f" ;;
      esac
      echo
done
```

So I'm confused on whether to state that both are supported but only one is the recommended one.